main_test.go

```diff
@@ -26,74 +26,124 @@ import (
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
+	"github.com/google/nftables"
 	"github.com/vishvananda/netns"
 )
 
-func Test_syncRules(t *testing.T) {
+type testSetup struct {
+	v4net   *net.IPNet
+	v6net   *net.IPNet
+	podnet  *net.IPNet
+	gwIface string
+	nftConn *nftables.Conn
+}
+
+func setupTest(t *testing.T) (setup testSetup, cleanup func()) {
+	t.Helper()
+
+	var origns, newns *netns.NsHandle
+
+	runtime.LockOSThread()
+	cleanup = func() {
+		if newns != nil {
+			newns.Close() // nolint:errcheck
+		}
+		if origns != nil {
+			_ = netns.Set(*origns)
+			origns.Close() // nolint:errcheck
+		}
+		runtime.UnlockOSThread()
+	}
+
 	if os.Getuid() != 0 {
+		cleanup()
```

Comment:
I think instead of calling `cleanup` you could use `t.Cleanup`. When you call it yourself, it is easy to forget about it somewhere.